src/main/java/gregtech/api/capability/impl/BoilerRecipeLogic.java

```diff
@@ -36,8 +41,10 @@ public class BoilerRecipeLogic extends AbstractRecipeLogic {
     private static final int FLUID_BURNTIME_TO_EU = 800 / FLUID_DRAIN_MULTIPLIER;
 
     private int currentHeat;
-    private int lastTickSteamOutput;
-    private int excessWater, excessFuel, excessProjectedEU;
+    private long lastTickSteamOutput;
+    private long excessWater;
+    private long excessFuel;
+    private long excessProjectedEU;
```

Comment:
Why are these values changed to longs? I don't really see a time where there will be more than max int of excess water produced by a recipe. Plus, keeping these as ints would allow you to get rid of some of the casts below.